list.go

```diff
@@ -193,6 +198,71 @@ func (db *RoseDB) LSet(key []byte, index int, value []byte) error {
 	return nil
 }
 
+// LRange returns the specified elements of the list stored at key.
+// The offsets start and stop are zero-based indexes, with 0 being the first element
+// of the list (the head of the list), 1 being the next element and so on.
+// These offsets can also be negative numbers indicating offsets starting at the end of the list.
+// For example, -1 is the last element of the list, -2 the penultimate, and so on.
+// If start is larger than the end of the list, an empty list is returned.
+// If stop is larger than the actual end of the list, Redis will treat it like the last element of the list.
+func (db *RoseDB) LRange(key []byte, start, end int) (values [][]byte, err error) {
+	db.listIndex.mu.RLock()
+	defer db.listIndex.mu.RUnlock()
+
+	if db.listIndex.trees[string(key)] == nil {
+		return nil, ErrKeyNotFound
+	}
+
+	db.listIndex.idxTree = db.listIndex.trees[string(key)]
+	// get List DataType meta info
+	headSeq, tailSeq, err := db.listMeta(key)
+	if err != nil {
+		return nil, err
+	}
+
+	var startSeq, endSeq uint32
+
+	// logical address to physical address
+	startSeq, err = db.listSequence(headSeq, tailSeq, start)
+	if err != nil {
+		return nil, err
+	}
+	endSeq, err = db.listSequence(headSeq, tailSeq, end)
+	if err != nil {
+		return nil, err
+	}
+	// normalize startSeq
+	if startSeq <= headSeq {
+		startSeq = headSeq + 1
+	}
+	if startSeq >= tailSeq {
```

Comment:
You can combine all conditions into one, that looks like more clear.
if startSeq >= tailSeq || endSeq <= headSeq || startSeq > endSeq {
    return nil, ErrWrongIndex
}